### PR TITLE
Initial Pull Request

### PR DESCRIPTION
**Pull request by obivanskiy**

### CODE REVIEW

File: PetProject/Protocols/HTTPRequestProperties.swift [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+//
+//  Created by Ivan Obodovskyi on 1/29/20.
+//  Copyright © 2020 Ivan Obodovskyi. All rights reserved.
+//
+
+import Foundation
+import Alamofire
+
+
+protocol HTTPRequestProperties {
```
Comment (savchukn) [inline PetProject/Protocols/HTTPRequestProperties.swift]: Naming: it is not properties, it is an object
Setter: why do we need setter here?
Optionals: interceptor & destination could be segregated to an optional protocol

File: PetProject/Protocols/NetworkProtocol.swift [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+//
+//  Created by Ivan Obodovskyi on 1/29/20.
+//  Copyright © 2020 Ivan Obodovskyi. All rights reserved.
+//
+
+import Foundation
+import Alamofire
+
+
+protocol NetworkProtocol {
```
Comment (savchukn) [inline PetProject/Protocols/NetworkProtocol.swift]: If this is NetworkClient why is is returning DownloadRequest? It is an object that only executing HTTP requests and it should return Rx wrapped response and error tuple

File: PetProject/SystemFiles/SceneDelegate.swift [hunk 3]
```diff
@@ -8,6 +8,7 @@
 
 import UIKit
 import SwiftUI
+import Alamofire
```
Comment (savchukn) [inline PetProject/SystemFiles/SceneDelegate.swift]: Should not be imported

File: PetProject/Protocols/NetworkDataEncoder.swift [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+//  Created by Ivan Obodovskyi on 1/29/20.
+//  Copyright © 2020 Ivan Obodovskyi. All rights reserved.
+//
+
+import Foundation
+import Alamofire
+
+
+protocol NetworkDataEncoder {
+    static var `default`: URLEncoding { get }
```
Comment (savchukn) [inline PetProject/Protocols/NetworkDataEncoder.swift]: What if it is a post request? Make it as an enum with most popular usage cases

File: PetProject/Networking/NetworkRequestObject.swift [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+//  PetProject
+//
+//  Created by Ivan Obodovskyi on 1/29/20.
+//  Copyright © 2020 Ivan Obodovskyi. All rights reserved.
+//
+
+import Foundation
+import Alamofire
+
+class RequestExampleObject: HTTPRequestProperties {
```
Comment (savchukn) [inline PetProject/Networking/NetworkRequestObject.swift]: Do we need it to be a class object?

Comment (obivanskiy) [reply PetProject/Networking/NetworkRequestObject.swift]: > Do we need it to be a class object?

Is it better to be a struct?

File: PetProject/Protocols/HTTPRequestProperties.swift (hunk 1 above)
Comment (obivanskiy) [reply PetProject/Protocols/HTTPRequestProperties.swift]: We need setter to be able to initialize RequestObject with certain properties. Is there any better way to do this?

**Pull request merged**